File: Server/Projeto_NFe.Infra.PDF/Features/Addressees/AddresseeModel.cs
using Projeto_NFe.Domain.Features.Addressees;
using Projeto_NFe.Domain.Utils;
using Projeto_NFe.Infra.PDF.Features.Addresses;

namespace Projeto_NFe.Infra.PDF.Features.Addressees
{
    public class AddresseeModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string CorporateName { get; set; }
        public string Cnpj { get; set; }
        public string Cpf { get; set; }
        public string StateRegistration { get; set; }
        public AddressModel Address { get; set; }
        public EnumPersonType PersonType { get; set; }

        public void Create(Addressee addressee)
        {
            Cnpj = addressee.Cnpj;
            Cpf = addressee.Cpf;
            Address = new AddressModel();
            Address.Neighborhood = addressee.Address.Neighborhood;
            Address.State = addressee.Address.State;
            Address.StreetName = addressee.Address.StreetName;
            Address.City = addressee.Address.City;
            Address.Number = addressee.Address.Number;
            Address.Country = addressee.Address.Country;
            Id = addressee.Id;
            StateRegistration = addressee.StateRegistration;
            Name = addressee.BusinessName;
            CorporateName = addressee.CorporateName;
            PersonType = addressee.PersonType;
        }
    }
}

File: Projeto_NFe/Client/src/app/features/addressee/shared/addressee.model.ts
import { PersonType } from './../../shared/person-type.enum';
import { AddressCommand } from './../../address/address.model';
export class Addressee {
    public id: number;
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public cpf: string;
    public stateRegistration: string;
    public streetName: string;
    public number: number;
    public neighborhood: string;
    public city: string;
    public state: string;
    public country: string;
    public personType: number;
}

export class AddresseeRegisterCommand {
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public cpf: string;
    public personType: string;
    public stateRegistration: string;
    public address: AddressCommand;

    constructor(addressee: any) {
        this.businessName = addressee.businessName;
        this.personType = addressee.personType;
        this.address = addressee.address;
        if (this.personType === PersonType.PERSON) {
            this.cpf = addressee.person.cpf;
        } else {
            this.cnpj = addressee.enterprise.cnpj;
            this.stateRegistration = addressee.enterprise.stateRegistration;
            this.corporateName = addressee.enterprise.corporateName;
        }
    }
}

export class AddresseeRemoveCommand {
    public addresseesId: number[] = [];

    constructor(selectedEntities: any) {
        this.addresseesId = selectedEntities.map((addressee: Addressee) => addressee.id);
    }
}
